### PR TITLE
[spirv] Add bit ops

### PR DESCRIPTION
**Pull request by denis0x0D**
This patch addresses an issue https://github.com/tensorflow/mlir/issues/173 and adds a few bit operations:

* OpBitFieldInsert
* OpBitFieldSExtract
* OpBitFieldUExtract

@antiagainst @MaheshRavishankar can you please take a look?
Thanks!

### CODE REVIEW

File: lib/Dialect/SPIRV/SPIRVOps.cpp [hunk 1]
```diff
@@ -733,6 +771,55 @@ static LogicalResult verify(spirv::BitcastOp bitcastOp) {
...
+}
+
+static LogicalResult verify(spirv::BitFieldInsertOp bitFieldOp) {
+  auto baseType = bitFieldOp.base()->getType();
+  auto insertType = bitFieldOp.insert()->getType();
+  auto resultType = bitFieldOp.getResult()->getType();
+
+  if ((baseType != insertType) || (baseType != resultType)) {
+    return bitFieldOp.emitError(
+               "expected the same type for the `base` operand, "
```
Comment (antiagainst) [inline lib/Dialect/SPIRV/SPIRVOps.cpp]: ... same type for base operand, insert operand, and result, ...

File: test/Dialect/SPIRV/Serialization/bit-ops.td [hunk 2]
```diff
@@ -6,6 +6,21 @@ spv.module "Logical" "GLSL450" {
     %0 = spv.BitCount %arg : i32
     spv.ReturnValue %0 : i32
   }
+  func @bit_field_insert(%base: vector<3xi32>, %insert: vector<3xi32>, %offset: i32, %count: i16) -> vector<3xi32> {
```
Comment (antiagainst) [inline test/Dialect/SPIRV/Serialization/bit-ops.td]: This file should have a .mlir suffix.

Comment (denis0x0D) [reply test/Dialect/SPIRV/Serialization/bit-ops.td]: oh, sorry about it!

**Comment by denis0x0D** [comment]
@antiagainst thanks a lot for review, updated!

**Pull request closed without merging**